### PR TITLE
[Map] Rework `MAP` creation method behavior when input is NULL

### LINKED ISSUE #11115: Maps can't handle NULL keys or values
### What happens?

```sql
SELECT MAP(NULL::INT[], [1, 2, 3]); 
--- Binder Error: Key and value list sizes don't match
SELECT MAP([1,2,3], NULL::INT[]);
--- Binder Error: Key and value list sizes don't match
SELECT MAP(NULL::INT[], NULL::INT[]);
--- results in {}, I would expect NULL
--- This only is apparent in the CLI version.
--- If I try this on shell.duckdb.org, you can't see a result as a result of unrelated bug https://github.com/duckdb/duckdb-wasm/issues/1671
--- As a workaround to that bug:
SELECT MAP(NULL::INT[], NULL::INT[]) IS NULL;
--- gives false, should be true
```

I would expect all of these to give me NULL

### To Reproduce

see above

### OS:

shell.duckdb.org

### DuckDB Version:

0.10.0

### DuckDB Client:

@duckdb/duckdb-wasm@1.28.1-dev166.0

### Full Name:

Nick Crews

### Affiliation:

Ship Creek Group

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #11115

### Old behavior:
`MAP(NULL, [1,2,3])` -> Error: Only LIST types are accepted arguments
`MAP([1,2,3], NULL)` -> Error: Only LIST types are accepted arguments
`MAP(NULL, NULL)` -> `{}` (empty map)
`MAP(NULL::INT[], [1,2,3])` -> Error: Key list can not be NULL
`MAP([1,2,3], NULL::INT[])` -> Error: Value list can not be NULL
`MAP(NULL::INT[], NULL::INT[])` -> Error: Key list can not be NULL

### New behavior:
`MAP(NULL, [1,2,3])` -> `NULL`
`MAP([1,2,3], NULL)` -> `NULL`
`MAP(NULL, NULL)` -> `NULL`
`MAP(NULL::INT[], [1,2,3])` -> `NULL`
`MAP([1,2,3], NULL::INT[])` -> `NULL`
`MAP(NULL::INT[], NULL::INT[])` -> `NULL`

### CODE REVIEW

File: src/core_functions/scalar/map/map.cpp [hunk 1]
```diff
@@ -87,22 +96,15 @@ static void MapFunction(DataChunk &args, ExpressionState &, Vector &result) {
...
 
 		auto keys_entry = keys_entries[keys_idx];
 		auto values_entry = values_entries[values_idx];
 
-		// validity checks
-		if (!keys_data.validity.RowIsValid(keys_idx)) {
-			MapVector::EvalMapInvalidReason(MapInvalidReason::NULL_KEY_LIST);
-		}
-		if (!values_data.validity.RowIsValid(values_idx)) {
-			MapVector::EvalMapInvalidReason(MapInvalidReason::NULL_VALUE_LIST);
```
Comment (Tishj) [inline src/core_functions/scalar/map/map.cpp:104]: Perhaps these MapInvalidReasons should be removed now

**Comment by Tishj** [comment]
It seems in the pandas analyzer we already didn't use this NULL_KEY_LIST invalid-reason because it got converted to a STRUCT instead

**Comment by NickCrews** [comment]
What should `TYPEOF(MAP_KEYS(MAP(NULL, [1,2,3])))` be?

**Comment by Tishj** [comment]
> What should `TYPEOF(MAP_KEYS(MAP(NULL, [1,2,3])))` be?

Different PR, I'll push that when this is merged as it depends on this

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**